### PR TITLE
testing app layer

### PR DESCRIPTION
**Pull request by colesanders**

### CODE REVIEW

File: apps/dashboard/src/app/lessons/lessons.component.spec.ts [hunk 1]
```diff
@@ -1,25 +1,83 @@
...
+    expect(spy).toHaveBeenCalled();
+  })
+
+  describe('should on save call lessonsFacade', () => {
+    it('updateLesson', () => {
+      const spy = jest.spyOn(lessonsFacade, 'updateLesson');
+
+      component.saveLesson(mockLesson);
+
+      expect(spy).toHaveBeenCalled();
```
Comment (joshdegouveia) [inline apps/dashboard/src/app/lessons/lessons.component.spec.ts:64]: You can make this test more specific by testing that the spy was called with the correct data. This is important if there is mapping logic within the method - but also good practice to use whenever you can. `toHaveBeenCalledWith(mockLesson)`

File: apps/dashboard/src/app/app.component.spec.ts [hunk 2]
```diff
@@ -1,19 +1,81 @@
...
+  });
+
   it('should create the app', () => {
-    const fixture = TestBed.createComponent(AppComponent);
-    const app = fixture.debugElement.componentInstance;
-    expect(app).toBeTruthy();
+    expect(component).toBeTruthy();
   });
+
+  it('should call logout on toolbar event-logout', () => {
```
Comment (joshdegouveia) [inline apps/dashboard/src/app/app.component.spec.ts:40]: There's no need to test the event handling on elements. Only test the component logic.

File: apps/dashboard/src/app/users/users.component.spec.ts [hunk 3]
```diff
@@ -1,25 +1,83 @@
...
+      ]
     })
     .compileComponents();
   }));
 
   beforeEach(() => {
     fixture = TestBed.createComponent(UsersComponent);
     component = fixture.componentInstance;
+    de = fixture.debugElement;
+    usersFacade = de.injector.get(UsersFacade);
```
Comment (joshdegouveia) [inline apps/dashboard/src/app/users/users.component.spec.ts:42]: Do not use the injector object to gain access to providers. Use the testbed.
`TestBed.get(UsersFacade)`.

File: jest.config.js [hunk 4]
```diff
@@ -5,5 +5,5 @@ module.exports = {
   },
   resolver: '@nrwl/jest/plugins/resolver',
   moduleFileExtensions: ['ts', 'js', 'html'],
-  coverageReporters: ['html'],
+  coverageReporters: ['html','text'],
```
Comment (joshdegouveia) [inline jest.config.js:8]: What does coverage does `text` provide?

Comment (colesanders) [reply jest.config.js:8]: Thats the modifier needed to add the coverage report to the command line. Otherwise it only provides the html reports.

**Pull request merged**